### DISCUSSION TITLE
New in v1.8.0 - ✨ Improved Diff Mode

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/codecompanion.nvim#150] [Announcements]
https://github.com/user-attachments/assets/76fbf5ac-c023-4fec-87cf-7290da3a9e4a

Thanks to everyone in #109 for their input on how we could improve the diff mode in CodeCompanion. Thanks to @S1M0N38 for his work on [dante.nvim](https://github.com/S1M0N38/dante.nvim) that I used heavily.

I also took this opportunity to improve the Inline module.

**Comment by SDGLBL** [comment]
I tried introducing a streaming diff similar to zed in neovim, and it seems this would be more suitable for the inline strategy.

![CleanShot 2024-09-01 at 00 09 53](https://github.com/user-attachments/assets/7b6855b0-ad66-46e1-b644-aa9b2f2a8831)
![CleanShot 2024-09-01 at 00 11 57](https://github.com/user-attachments/assets/e7f7e18a-afdb-4120-883b-203c579a33e5)
![CleanShot 2024-09-01 at 00 15 28](https://github.com/user-attachments/assets/15025bd4-c9bb-4cee-a06d-f0f844507be3)

**Reply by S1M0N38** [reply]
I agree. I'm experimenting with [mini.diff](https://github.com/echasnovski/mini.diff) to highlight the diff. It seems handy
